aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/states/SignInChallengeState.kt

```diff
@@ -58,8 +59,25 @@ internal sealed class SignInChallengeState : State {
                 }
                 is Verifying -> when (challengeEvent) {
                     is SignInChallengeEvent.EventType.Verified -> StateResolution(Verified())
+                    is SignInChallengeEvent.EventType.ThrowError -> {
+                        val action = challengeActions.resetToWaitingForAnswer(challengeEvent, challengeEvent.challenge)
+                        StateResolution(Error(challengeEvent.exception, challengeEvent.challenge), listOf(action))
+                    }
+
                     else -> defaultResolution
                 }
+                is Error -> {
+                    when (challengeEvent) {
+                        is SignInChallengeEvent.EventType.VerifyChallengeAnswer -> {
+                            val action = challengeActions.verifyChallengeAuthAction(challengeEvent, oldState.challenge)
+                            StateResolution(Verifying(oldState.challenge.challengeName), listOf(action))
+                        }
+                        is SignInChallengeEvent.EventType.WaitForAnswer -> {
```

Comment:
Have we discussed with iOS. iOS Does not have this resolution.

Reply:
I have verified this with iOS and get their blessing. In iOS the SM does not work the exact same way. In Android when we send an event and then listen to the state machine if the current SignInChallengeSM is in Error state it will error out right away. This way we can ensure this will not error out and go to WaitForAnswer which is the right state to be in when the customer tries the code again. To make this more robust I can null out the exception inside Error state and then check for that null inside the plugin to ensure we do not duplicate error responses in cases of concurrent calls. (I will do that in the next revision).

Reply:
Done. I have also remove the resetToWaitingState as the null guard is working well.

Reply:
This looks like no-op. Since it is same as transition to default resolution.